src/Converter/IntToRoman.php

```diff
@@ -38,17 +41,100 @@ final class IntToRoman
 		'I' => 1,
 	];
 
+	/** @var string[] */
+	private static $fractionConversionTable = [
+		'·',
+		'··',
+		'···',
+		'····',
+		'·····',
+		'S',
+		'S·',
+		'S··',
+		'S···',
+		'S····',
+		'S·····',
+	];
+
 
 	/**
 	 * @param BigNumber|int|string|Stringable $input
 	 * @return string
+	 * @throws OutOfRomanNumberSetException
 	 */
 	public static function convert($input): string
 	{
-		$int = BigInteger::of((string) $input);
+		try {
+			$integer = BigInteger::of((string) $input);
+		} catch (RoundingNecessaryException $e) {
+			return self::convertRationalNumber($input);
+		}
+
+		return self::convertInteger($integer);
+	}
+
+
+	/**
+	 * @param BigNumber|int|string|Stringable $input
+	 * @return string
+	 * @throws OutOfRomanNumberSetException
+	 * @see https://en.wikipedia.org/wiki/Roman_numerals (Fractions)
+	 */
+	public static function convertRationalNumber($input): string
+	{
+		$rationalNumber = BigRational::of((string) $input)->simplified();
+
+		if ($rationalNumber->isLessThan('1/12')) {
+			throw new OutOfRomanNumberSetException((string) $input);
+		}
+
+		$denominatorOriginal = $rationalNumber->getDenominator();
+
+		if (in_array((string) $denominatorOriginal, ['2', '3', '4', '6', '12'], true) && $denominatorOriginal->isLessThanOrEqualTo(12)) {
```

Comment:
I think better is to use array of primary numbers smaller than `sqrt($input)`.

Reply:
Oh, now I understand, because it is this table:

<img width="723" alt="Snímek obrazovky 2020-06-05 v 9 24 37" src="https://user-images.githubusercontent.com/4738758/83848585-612de580-a70e-11ea-98a6-c47fcf459fd9.png">

I think your code looks so magical. It can be better to use exception table (array with `inputFraction => romanNumeral`) and call it by key.

What do you think about?

Reply:
> I think your code looks so magical. It can be better to use exception table (array with `inputFraction => romanNumeral`) and call it by key.

Ok, I'll implement it this way.

Reply:
@janbarasek Is it more clear now?

Reply:
Yes, it's looks soo clear! :)